### PR TITLE
Add support to Shakapacker

### LINKED ISSUE #156: Support Shakapacker 8
Shakapacker 8 removed `Webpacker` spelling ref: shakacode/shakapacker#429

This is preventing CI to pass: https://github.com/diowa/icare/pull/1725

I'm working on a PR

### PR DESCRIPTION
**Pull request by tagliala**
Shakapacker is the official, actively maintained successor to Webpacker.

Shakapacker v7 changed the spelling of `Webpacker` to `Shakapacker` in
the entire project, but still provided backward compatibility for
`Webpacker` spelling.

v8 dropped the deprecated spelling

This commit also:
- Checks if `Shakapacker` is defined; if not, it falls back on
  `Webpacker`.
- Uses the scope resolution operator to resolve at top-level
  scope
- Checks `protocol` instead of `https?` because the former is available
  from Webpacker 3 and the latter is not available anymore in
  Shakapacker >= 8

Refs:
- shakacode/shakapacker#414
- shakacode/shakapacker#429
- shakacode/shakapacker#486

Close jamesmartin#156

### CODE REVIEW

File: lib/inline_svg/webpack_asset_finder.rb
Comment (paypro-leon) [inline lib/inline_svg/webpack_asset_finder.rb:51]: We also would like to see Shakapacker 8 support.

`Shakapacker::DevServer` has a `protocol` method which can be `http` or `https`. This could be used for Shakapacker 8 to set `use_ssl`

Comment (tagliala) [reply lib/inline_svg/webpack_asset_finder.rb:51]: I would propose if it is possible to re-introduce the `https?` predicate on Shakapacker

Comment (tagliala) [reply lib/inline_svg/webpack_asset_finder.rb:51]: Let's wait for feedback here: shakacode/shakapacker#486

Comment (paypro-leon) [reply lib/inline_svg/webpack_asset_finder.rb:51]: I agree that would be the nicest solution, but since `8.0.0` is already released that won't work for that particular version. In our own codebase we actually monkey patch that back, which `inline_svg` would also have to do.

Comment (tagliala) [reply lib/inline_svg/webpack_asset_finder.rb]: Switched to `protocol` for both webpacker and shakapacker, since it is available from v3

**Comment by paypro-leon** [comment]
@tagliala Looks good to me. Just need a maintainer to have a look and review.

**Comment by tagliala** [comment]
Thanks.

I have two successful builds on Shakapacker 7 and 8 at:
- https://github.com/diowa/icare/actions/runs/9256714063
- https://github.com/diowa/icare/actions/runs/9256838181

A proper solution would be to create a new app with Shakapacker 8 and make an integration test here, as per #159, but this would require to fix the test apps repo

**Comment by tagliala** [comment]
I have a working build against Rails 7.1 / Shakapacker 8: https://github.com/tagliala/inline_svg/actions/runs/9549035858

I've removed the old test against webpacker because of python 2.7 and deprecations.

Please let me know if you need some help or if I should fork

**Comment by jamesmartin** [comment]
> Please let me know if you need some help or if I should fork

Thanks for your work so far. Sorry for the delayed responses, I was away on vacation and then work trips. Let's get CI working properly and then get this merged.

**Comment by tagliala** [comment]
Hi,

> Let's get CI working properly and then get this merged.

I did not invest time in webpacker (it should have been upgraded from v4 to v5 in the rails 6 branch), so this is expected to fail against the CI like it is failing now

#159 is based on this PR and should make the CI pass agains Rails 7.1 / Shakapacker 8

If you are interested, I can make a PR to drop compatibility with legacy Rails and Ruby versions

**Comment by jamesmartin** [comment]
> If you are interested, I can make a PR to drop compatibility with legacy Rails and Ruby versions

Sounds good. Let's do that. 👍

**Comment by tagliala** [comment]
Friendly bump.

The first step would be to merge

- jamesmartin/inline_svg_test_app#24
- jamesmartin/inline_svg_test_app#25
- jamesmartin/inline_svg_test_app#26

Then this PR can be merged because it is not breaking.

Then, after a change to the target repo in actions, it would be possible to merge:
- jamesmartin/inline_svg#159

To have a working CI.

At that point it would be possible to release a minor version with proper Shakapacker support.

After the minor release, it would be possible to consider to drop old Ruby and Rails support and go for 2.0

**Comment by jamesmartin** [comment]
Thanks, this is on my TODO list.

**Comment by tagliala** [comment]
Thanks, please let me know if you need help and what strategy you decide, everything works for me

**Review by G-Rath — APPROVED** [review]
looking good to me - we're using this gem on [nzsl-share](https://github.com/ackama/nzsl-share), and [this change is working well with our Shakapacker v8 upgrade](https://github.com/ackama/nzsl-share/pull/672).

(also hi @tagliala! appreciate your work on this)

**Review by jamesmartin — APPROVED** [review]

**Comment by jamesmartin** [comment]
Thanks for your patience, @tagliala. We're pushing forward with the [plan to properly support Shakapacker v8](https://github.com/jamesmartin/inline_svg/issues/156#issuecomment-2325470659) moving forward and also to drop support for end-of-life Rails versions in an incoming v2.0 of this gem.

**Pull request merged**
